mmv1/third_party/terraform/services/dataflow/resource_dataflow_job.go.tmpl

```diff
@@ -51,6 +51,37 @@ func resourceDataflowJobLabelDiffSuppress(k, old, new string, d *schema.Resource
 	return false
 }
 
+// ResourceDataflowJobTemplateGcsPathDiffSuppress suppresses diffs on the template_gcs_path field
+// when the only diff is the region name in the bucket.
+// `old` is the value set by service and `new` is what is stored in the config. When using a public
+// template bucket, it will be automatically turned to use a regional bucket path. In this case,
+// we want to supress the diff.
+// Example:
+//   - old: gs://template-bucket-us-central1/path/to/file.txt
+//   - new: gs://template-bucket/path/to/file.txt
+func ResourceDataflowJobTemplateGcsPathDiffSuppress(k, old, new string, d *schema.ResourceData) bool {
```

Comment:
`resourceDataflowJobTypeCustomizeDiff` forces the resource recreation (makes every field `forcenew`) if any non-virtual fields (except `terraform_labels` field) changed for batch jobs, we need to extend it with this diff suppress logic. 

A working example is available in https://github.com/GoogleCloudPlatform/magic-modules/pull/13351, but the code can be optimized. Hope this helps

Reply:
Done.